forward_engineering/services/statements/indexesStatements.js

```diff
@@ -10,7 +10,7 @@ const { INDEX_TYPE } = require('../../../shared/enums/n1ql');
  */
 const getIndexesScript = ({ namespace, bucketName, scopeName, collectionName, indexes, properties }) => {
 	const collectionIndexes = indexes ?? [];
-	const keyIdToName = getIndexKeyIdToKeyNameMap(properties);
+	const keyIdToName = properties ? getIndexKeyIdToKeyNameMap(properties) : {};
```

Comment:
Why not 'upgrading' the `getIndexKeyIdToKeyNameMap` to provide the similar fallback there, instead of trying to deal with the consequences?